File: packages/appwsgi/appwsgi%20918.tar.bz2/appwsgi/www/register-admin/register-admin.py
# Copyright(c) gert.cuykens@gmail.com

from xmlframe import xmlframe

def application(environ, response):
    sid = environ.get('HTTP_COOKIE','')
    doc = environ['wsgi.input'].read(int(environ.get('CONTENT_LENGTH','0')))
    data,size,sid = xmlframe(doc,sid,any)
    response('200 OK',[('Content-type', 'text/xml'),('Content-Length', size),('Set-Cookie', sid)])
    return [data]

def any(db,v):
    if   v['cmd']=='update': db.execute("UPDATE register SET email=%s, name=%s, adress=%s, city=%s, country=%s, phone=%s, gid=%s WHERE uid=%s",(v['email'],v['name'],v['adress'],v['city'],v['country'],v['phone'],v['gida'],v['uida']))
    elif v['cmd']=='delete': db.execute("DELETE FROM register WHERE uid=%s",(v['uida']))
    elif v['cmd']=='select': db.execute("SELECT email,name,adress,city,country,phone,uid,gid FROM register WHERE email LIKE %s AND name LIKE %s AND adress LIKE %s AND city LIKE %s AND country LIKE %s AND phone LIKE %s OR gid=%s OR uid=%s",('%'+v['email']+'%','%'+v['name']+'%','%'+v['adress']+'%','%'+v['city']+'%','%'+v['country']+'%','%'+v['phone']+'%',v['gida'],v['uida']))
    

File: packages/appwsgi/appwsgi%20667.tar.bz2/trunk/www/register-admin/register-admin.js
// Copyright(c) gert.cuykens@gmail.com
registeradmin=
{

 'select':function()
 {
  http.fml=function(xml)
  {
   var c
   if(c=document.getElementById('cregister'))while(c.lastChild)c.removeChild(c.lastChild)
   register=
   {
    'id':'register',
    'onselect':function()
    {
     document.getElementById('email').value=register.selection[1]
     document.getElementById('name').value=register.selection[2]
     document.getElementById('adress').value=register.selection[3]
     document.getElementById('city').value=register.selection[4]
     document.getElementById('country').value=register.selection[5]
     document.getElementById('phone').value=register.selection[6]
     document.getElementById('uid').value=register.selection[7]
     document.getElementById('gid').value=register.selection[8]
    }
   }
   dg.init(register,xml)
  }
  http.xml='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <cmd>select</cmd>\n'
  http.xml+=' <email>'+document.getElementById('email').value+'</email>\n'
  http.xml+=' <name>'+document.getElementById('name').value+'</name>\n'
  http.xml+=' <adress>'+document.getElementById('adress').value+'</adress>\n'
  http.xml+=' <city>'+document.getElementById('city').value+'</city>\n'
  http.xml+=' <country>'+document.getElementById('country').value+'</country>\n'
  http.xml+=' <phone>'+document.getElementById('phone').value+'</phone>\n'
  http.xml+=' <uida>'+document.getElementById('uid').value+'</uida>\n'
  http.xml+=' <gida>'+document.getElementById('gid').value+'</gida>\n'
  http.xml+='</root>'
  http.send()
 },
 
 'update':function()
 {
  http.fml=function(xml)
  {
   if(c=document.getElementById('cregister'))while(c.lastChild)c.removeChild(c.lastChild)
  }
  http.xml='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <cmd>update</cmd>\n'
  http.xml+=' <email>'+document.getElementById('email').value+'</email>\n'
  http.xml+=' <name>'+document.getElementById('name').value+'</name>\n'
  http.xml+=' <adress>'+document.getElementById('adress').value+'</adress>\n'
  http.xml+=' <city>'+document.getElementById('city').value+'</city>\n'
  http.xml+=' <country>'+document.getElementById('country').value+'</country>\n'
  http.xml+=' <phone>'+document.getElementById('phone').value+'</phone>\n'
  http.xml+=' <uida>'+document.getElementById('uid').value+'</uida>\n'
  http.xml+=' <gida>'+document.getElementById('gid').value+'</gida>\n'
  http.xml+='</root>'
  http.send()
 },

 'remove':function()
 {
  http.fml=function(xml)
  {
   if(c=document.getElementById('cregister'))while(c.lastChild)c.removeChild(c.lastChild)
   document.getElementById('email').value=''
   document.getElementById('name').value=''
   document.getElementById('adress').value=''
   document.getElementById('city').value=''
   document.getElementById('country').value=''
   document.getElementById('phone').value=''
   document.getElementById('uid').value=''
   document.getElementById('gid').value=''
  }
  http.xml='<?xml version="1.0" encoding="UTF-8"?>\n'
  http.xml+='<root>\n'
  http.xml+=' <cmd>delete</cmd>\n'
  http.xml+=' <uida>'+document.getElementById('uid').value+'</uida>\n'
  http.xml+='</root>'
  http.send()
 }

}

